In [1]:
import pandas as pd
import statsmodels.api as sm


In [2]:
df = pd.read_csv('../data/coffeeleaflet.csv')
df.head()

grade  sex   age   mob  bulgnationality  langbulg  mumage  mumedu  mumprof  \
0     10  1.0  16.0  11.0                1         1    42.0     2.0      2.0   
1      9  1.0  15.0   5.0                1         1    38.0     2.0      4.0   
2     10  1.0  15.0  12.0                1         1    42.0     2.0      7.0   
3      9  1.0  15.0   4.0                1         1    40.0     3.0      7.0   
4     12  0.0  18.0   6.0                1         1    44.0     2.0      1.0   

   dadage  ...  awarepesticide  awaredeforestation  awarewastewater  \
0    44.0  ...             3.0                 1.0              3.0   
1    44.0  ...             3.0                 2.0              3.0   
2    45.0  ...             1.0                 2.0              1.0   
3    43.0  ...             1.0                 2.0              3.0   
4    44.0  ...             1.0                 3.0              1.0   

   awarebiodiversityloss  awareunfairworking  reusepurposeful  reusesoil  \
0                    1.0                 1.0              1.0        1.0   
1                    1.0                 2.0              2.0        3.0   
2                    3.0                 4.0              1.0        1.0   
3                    4.0                 3.0              2.0        1.0   
4                    4.0                 1.0              1.0        1.0   

   choiceprice  choicetastepleasure  choiceenvironsocial  
0          6.0                  6.0                  6.0  
1          2.0                  5.0                  1.0  
2          6.0                  6.0                  6.0  
3          6.0                  6.0                  6.0  
4          1.0                  4.0                  3.0  

[5 rows x 48 columns]

In [3]:
df= df.loc[:, ['treatment', 'awarewaste']]
df = df.dropna()
D = df['treatment']
D = sm.add_constant(D) # add intercept
Y = df['awarewaste']
results = sm.OLS(Y, D).fit(cov_type='HC0')
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:             awarewaste   R-squared:                       0.017
Model:                            OLS   Adj. R-squared:                  0.015
Method:                 Least Squares   F-statistic:                     8.864
Date:                Thu, 25 Dec 2025   Prob (F-statistic):            0.00305
Time:                        22:22:55   Log-Likelihood:                -742.26
No. Observations:                 504   AIC:                             1489.
Df Residuals:                     502   BIC:                             1497.
Df Model:                           1                                         
Covariance Type:                  HC0                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.0079      0.063     32.001      0.0

## Intrepretation of A/B Test Results
In this example, we conduct an A/B test to evaluate the impact of a treatment (e.g., a marketing leaflet) on the awareness of waste among participants. We use the `statsmodels` library to perform an Ordinary Least Squares (OLS) regression analysis.

The treatment coefficient of 0.2801 means that being in the treatment group (receiving the leaflet) is associated with a 28.01% increase in the awareness of waste compared to the control group (not receiving the leaflet). This suggests that the leaflet had a positive effect on raising awareness about waste management among the participants.

Given an intercept of 2.01, this indicates that, on average, participants in the control group had an awareness score of 2.01. Therefore, the treatment increased this score to approximately 2.29 (2.01 + 0.2801) for those in the treatment group.